bravado/client.py

```diff
@@ -102,11 +102,23 @@ def from_url(cls, spec_url, http_client=None, request_headers=None,
         :param config: Config dict for bravado and bravado_core.
             See CONFIG_DEFAULTS in :module:`bravado_core.spec`.
             See CONFIG_DEFAULTS in :module:`bravado.client`.
+
+        :rtype: :class:`bravado_core.spec.Spec`
         """
         log.debug(u"Loading from %s" % spec_url)
         http_client = http_client or RequestsClient()
         loader = Loader(http_client, request_headers=request_headers)
         spec_dict = loader.load_spec(spec_url)
+
+        # RefResolver may have to download additional json files (remote refs)
+        # via http. Wrap http_client's request() so that request headers are
+        # passed along with the request transparently. Yeah, this is not ideal,
+        # but since RefResolver has new found responsibilities, it is
+        # functional.
```

Comment:
I was thinking more like unwrapping the wrapper once the work is finished (after the call to `bravado-core`, which guarantees that the work of wrapper is done), instead of having the check looking for `operation` key (which looks more hacky to me). I am not too sure on its implementation details but can think on that tomo.